internal/renderers/skia/vulkan_surface.rs

```diff
@@ -0,0 +1,427 @@
+        self.recreate_swapchain.set(true);
+        Ok(())
+    }
+
+    fn render(
+        &self,
+        size: PhysicalWindowSize,
+        callback: impl FnOnce(&mut skia_safe::Canvas, &mut skia_safe::gpu::DirectContext),
+    ) -> Result<(), i_slint_core::platform::PlatformError> {
+        let gr_context = &mut self.gr_context.borrow_mut();
+
+        let device = self.device.clone();
+
+        self.previous_frame_end.borrow_mut().as_mut().unwrap().cleanup_finished();
+
+        if self.recreate_swapchain.take() {
+            let mut swapchain = self.swapchain.borrow_mut();
+            let (new_swapchain, new_images) = swapchain
+                .recreate(SwapchainCreateInfo {
+                    image_extent: [size.width, size.height],
+                    ..swapchain.create_info()
+                })
+                .map_err(|vke| format!("Error re-creating Vulkan swap chain: {vke}"))?;
+
+            *swapchain = new_swapchain;
+
+            let mut new_swapchain_image_views = Vec::with_capacity(new_images.len());
+
+            for image in &new_images {
+                new_swapchain_image_views.push(ImageView::new_default(image.clone()).map_err(
+                    |vke| format!("fatal: Error creating image view for swap chain image: {vke}"),
+                )?);
+            }
+
+            *self.swapchain_images.borrow_mut() = new_images;
+            *self.swapchain_image_views.borrow_mut() = new_swapchain_image_views;
+        }
+
+        let swapchain = self.swapchain.borrow().clone();
+
+        let (image_index, suboptimal, acquire_future) =
+            match vulkano::swapchain::acquire_next_image(swapchain.clone(), None) {
+                Ok(r) => r,
+                Err(AcquireError::OutOfDate) => {
+                    self.recreate_swapchain.set(true);
+                    return Ok(()); // Try again next frame
+                }
+                Err(e) => return Err(format!("Vulkan: failed to acquire next image: {e}").into()),
+            };
+
+        if suboptimal {
+            self.recreate_swapchain.set(true);
+        }
+
+        let width = swapchain.image_extent()[0];
+        let width: i32 = width
+            .try_into()
+            .map_err(|_| format!("internal error: invalid swapchain image width {width}"))?;
+        let height = swapchain.image_extent()[1];
+        let height: i32 = width
+            .try_into()
+            .map_err(|_| format!("internal error: invalid swapchain image height {height}"))?;
+
+        let image_view = self.swapchain_image_views.borrow()[image_index as usize].clone();
+        let image_object = image_view.image();
+
+        let format = image_view.format();
+
+        let (vk_format, color_type) = match format {
+            Some(vulkano::format::Format::B8G8R8A8_UNORM) => {
+                (skia_safe::gpu::vk::Format::B8G8R8A8_UNORM, skia_safe::ColorType::BGRA8888)
+            }
+            _ => unimplemented!(),
```

Comment:
We can't run into this without the  constructor failing first. But I can change this into a debug_assert.